deploy/us-east-2-dev/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: storetheindex
+spec:
+  replicas: 1
+  serviceName: storetheindex
+  selector:
+    matchLabels:
+      app: storetheindex
+      release: storetheindex-0
+  volumeClaimTemplates:
+  template:
+    spec:
+      securityContext:
+        fsGroup: 532
+      volumes:
+        - name: storetheindex-identity-volume
+          secret:
+            secretName: storetheindex-identity
+            defaultMode: 0600
+        - name: storetheindex-config-volume
+          configMap:
+            name: storetheindex-0-storetheindex-config
+        - name: shared-volume
+          persistentVolumeClaim:
+            claimName: storetheindex-pvc
+      initContainers:
+        - args:
+          - -c
+          - envsubst </config/config >/data/storetheindex/config
+          command:
+          - bash
+          env:
+          - name: STORETHEINDEX_PRIVATE_KEY
+            valueFrom:
+              secretKeyRef:
+                key: privkey
+                name: storetheindex-identity
+          - name: STORETHEINDEX_PEER_ID
+            valueFrom:
+              secretKeyRef:
+                key: peerid
+                name: storetheindex-identity
+          image: cmd.cat/gettext
+          name: setupconfig
+          volumeMounts:
+          - mountPath: /data/storetheindex
+            name: shared-volume
+            subPath: ntwk-mainnet-storetheindex/storetheindex-0
+          - mountPath: /config
+            name: storetheindex-config-volume
+      containers:
+        - name: application
```

Comment:
What's the point of this container? You now have two containers that are almost identical in the compiled yaml file

Reply:
This is the main application container, i.e. the container that runs the daemon. The previous one is used to basically run `sit init` as an `initContainer`. I think we can remove the init container entirely and use mounted pre-initialised config pointed out [here](https://github.com/filecoin-project/storetheindex/pull/242/files/b9b880b4fb019405b83ecf4fefaa81db0fe28fd2#r815721055).

Reply:
This is the compiled yaml:
https://gist.github.com/MarcoPolo/cf861f5fd6d46bb694ffcaa4ea2daa92

Notice you have two identical containers here ([named application](https://gist.github.com/MarcoPolo/cf861f5fd6d46bb694ffcaa4ea2daa92#file-full-yaml-L183)) and here ([named storetheindex](https://gist.github.com/MarcoPolo/cf861f5fd6d46bb694ffcaa4ea2daa92#file-full-yaml-L203)). This is completely independent of the initContainer defined [here](https://gist.github.com/MarcoPolo/cf861f5fd6d46bb694ffcaa4ea2daa92#file-full-yaml-L214).

My question is why is there both a container named application and another one named storetheindex.

Reply:
I totally misunderstood your question; You are right, that's not clear for me either :)